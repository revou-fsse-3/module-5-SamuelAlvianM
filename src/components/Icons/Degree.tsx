const Degree = ({ temp }: { temp: number }): JSX.Element => (
  <>
    <span role="degree">
      {temp}
      <sup>o</sup>C
    </span>
  </>
)

export default Degree
