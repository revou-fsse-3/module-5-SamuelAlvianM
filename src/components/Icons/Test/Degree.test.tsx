import React from 'react';
import { render, screen } from '@testing-library/react';
import Degree from '../Degree';
import {test, expect, describe} from 'vitest';


test('Degree component renders correctly', () => {
  const { getByRole} = render(<Degree temp={25} />);
  const degreeElement = screen.getByRole('degree');
  expect(degreeElement).toMatchSnapshot();
});