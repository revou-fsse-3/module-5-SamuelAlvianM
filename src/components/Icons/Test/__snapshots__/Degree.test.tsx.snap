// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Degree component renders correctly 1`] = `
<span
  role="degree"
>
  25
  <sup>
    o
  </sup>
  C
</span>
`;
